starknet_py/tests/e2e/client/client_test.py

```diff
@@ -128,38 +121,30 @@ async def test_get_transaction_receipt(
     assert receipt.block_number == block_with_invoke_number
 
 
-@pytest.mark.parametrize(
-    "transaction",
-    [
-        Invoke(
-            contract_address=0x1,
-            entry_point_selector=get_selector_from_name("increase_balance"),
+@pytest.mark.asyncio
+async def test_estimate_fee_invoke(account, contract_address):
+    invoke_tx = await account.sign_invoke_transaction(
+        calls=Call(
+            to_addr=contract_address,
+            selector=get_selector_from_name("increase_balance"),
             calldata=[123],
-            max_fee=0,
-            version=0,
-            signature=[0x0, 0x0],
-            nonce=None,
         ),
-        Declare(
-            contract_class=create_compiled_contract(
-                compiled_contract=read_contract("map_compiled.json")
-            ),
-            sender_address=0x1,
-            max_fee=0,
-            signature=[0x0, 0x0],
-            nonce=0,
-            version=0,
-        ),
-    ],
-)
-@pytest.mark.asyncio
-async def test_estimate_fee(transaction, contract_address, client):
-    if isinstance(transaction, Invoke):
-        transaction = dataclasses.replace(
-            transaction, contract_address=contract_address
-        )
+        max_fee=MAX_FEE,
+    )
+    invoke_tx = await account.sign_for_fee_estimate(invoke_tx)
+    estimate_fee = await account.client.estimate_fee(tx=invoke_tx)
+
+    assert isinstance(estimate_fee.overall_fee, int)
+    assert estimate_fee.overall_fee > 0
 
-    estimate_fee = await client.estimate_fee(tx=transaction, block_number="latest")
+
+@pytest.mark.asyncio
+async def test_estimate_fee_declare(account):
```

Comment:
Are we testing `deploy_account` fee estimation somewhere?

Reply:
Yes, just below this test.